pkg/storage/db.go

```diff
@@ -320,11 +320,17 @@ func (db *DB) GetPartitions(ctx context.Context) ([]domain.Partition, error) {
 			End:       end,
 		})
 	}
-
 	if err := rows.Close(); err != nil {
 		return nil, err
 	}
-
+	for _, v := range partitions {
+		row := db.sqldb.QueryRow("SELECT count(*) FROM " + v.Name)
```

Comment:
suggestion: as the number of partitions increase over time, we will pay a cost of additional round trips. You don't need to make this change at the moment, but it may be worth getting a baseline for the performance of using  multiple queries, and comparing it agains building up one query to execute: https://stackoverflow.com/a/10185612